### PR TITLE
CAS-1065 [Docs] Restructure ChannelsHeader component docs

### PR DESCRIPTION
**Pull request by samiuelson**
https://stream-io.atlassian.net/browse/CAS-1065

🎯 Goal
Restructuring ChannelListHeader component docs

🧪 Testing
* Run docs locally `npx stream-chat-docusaurus -i -s`
* Open Channel List Header component section and read it.

☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### CODE REVIEW

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 1]
```diff
@@ -1,17 +1,50 @@
 # Channel List Header
 
-<!-- TODO: Import whatever makes sense to import from https://getstream.io/chat/docs/android/channel_list_header_view/?language=kotlin -->
-
 ## Overview
 
-<!-- TODO: Brief description and a couple screenshots with default styling. -->
+`ChannelListHeaderView` is a component designed to be displayed at the channels screen of your app.
```
Comment (leandroBorgesFerreira) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:5]: I see where I should put the View, but not what it does, which I fell that is missing here. Something like: "This is the view that shows the connection status, the title of the channel or conversation name..." would be nice.

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 2]
```diff
@@ -1,17 +1,50 @@
 # Channel List Header
 
-<!-- TODO: Import whatever makes sense to import from https://getstream.io/chat/docs/android/channel_list_header_view/?language=kotlin -->
-
 ## Overview
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:3]:
```suggestion
```

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md (hunk 1 above)
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:5]:
```suggestion
`ChannelListHeaderView` is a component designed to be displayed on the channels screen of your app. 
```

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 3]
```diff
@@ -1,17 +1,50 @@
...
+// Optional: setup listeners for user avatar and action button 
+chanelHeaderView.setOnActionButtonClickListener {
+    navigateSafely(R.id.action_homeFragment_to_addChannelFragment)
+}
+chanelHeaderView.setOnUserAvatarClickListener {
+    binding.drawerLayout.openDrawer(GravityCompat.START)
+}
+```
+All the logic of subscribing to data emitted by view model is provided by the `ChannelListHeaderViewModel::bindView` function. 
+By default, view model will make view display avatar of the currently logged-in user and call the view to display "Searching for network" state when needed.
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:39]:
```suggestion
By default, the ViewModel will make the view display avatar of the currently logged-in user and call the view to display "Searching for network" state when needed.
```

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 4]
```diff
@@ -1,17 +1,50 @@
...
+<io.getstream.chat.android.ui.channel.list.header.ChannelListHeaderView
+    android:id="@+id/channelListHeaderView"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    />
+```
 
-<!-- TODO: This section -->
+`ChannelListHeaderView` is supposed to work with `ChannelListHeaderViewModel`. 
+The basic setup of the view model and connecting it with the view can be done in the following way:
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:24]:
```suggestion
The basic setup of the ViewModel and connecting it with the view can be done in the following way:
```

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 5]
```diff
@@ -20,22 +53,29 @@ channelListHeaderView.setOnUserAvatarClickListener {
...
-        app:layout_constraintTop_toTopOf="parent"
-        app:streamUiShowActionButton="false"
+        app:streamUiShowUserAvatar="false"
+        app:streamUiShowOfflineProgressBar="false"
+        app:streamUiActionButtonIcon="@drawable/ic_stream_logo"
+        app:streamUiOnlineTitleTextStyle="bold"
         />
 ```
+
+The example above, hides the avatar view, makes the title text bold and sets the drawable of the action button to a drawable specified.
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:75]:
```suggestion
The example above hides the avatar view, makes the title text bold and sets the drawable of the action button to a drawable specified.
```

**Review by adasiewiczr — APPROVED** [review]
LGTM but let's wait for a second review from @nikomancy or @zsmb13

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md (hunk 1 above)
Comment (samiuelson) [reply docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md]: Explained this more precisely

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 6]
```diff
@@ -1,17 +1,51 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    />
+```
 
-<!-- TODO: This section -->
+`ChannelListHeaderView` is supposed to work with `ChannelListHeaderViewModel`. 
+The basic setup of the ViewModel and connecting it with the view can be done in the following way:
+ ```kotlin
+// Instantiate the view model
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:27]:
```suggestion
// Instantiate the ViewModel
```

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 7]
```diff
@@ -1,17 +1,51 @@
...
+    />
+```
 
-<!-- TODO: This section -->
+`ChannelListHeaderView` is supposed to work with `ChannelListHeaderViewModel`. 
+The basic setup of the ViewModel and connecting it with the view can be done in the following way:
+ ```kotlin
+// Instantiate the view model 
+val channelListHeaderViewModel: ChannelListHeaderViewModel by viewModels()
+// Bind the view model with ChannelListView
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:29]:
```suggestion
// Bind the ViewModel with ChannelListView 
```

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 8]
```diff
@@ -1,17 +1,51 @@
...
+channelListHeaderViewModel.bindView(chanelHeaderView, viewLifecycleOwner)
+// Optional: setup listeners for user avatar and action button 
+chanelHeaderView.setOnActionButtonClickListener {
+    navigateSafely(R.id.action_homeFragment_to_addChannelFragment)
+}
+chanelHeaderView.setOnUserAvatarClickListener {
+    binding.drawerLayout.openDrawer(GravityCompat.START)
+}
+```
+All the logic of subscribing to data emitted by view model is provided by the `ChannelListHeaderViewModel::bindView` function.
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:39]:
```suggestion
All the logic of subscribing to data emitted by the ViewModel is provided by the `ChannelListHeaderViewModel::bindView` function. 
```

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 9]
```diff
@@ -1,17 +1,51 @@
...
+`ChannelListHeaderView` is supposed to work with `ChannelListHeaderViewModel`. 
+The basic setup of the ViewModel and connecting it with the view can be done in the following way:
+ ```kotlin
+// Instantiate the view model 
+val channelListHeaderViewModel: ChannelListHeaderViewModel by viewModels()
+// Bind the view model with ChannelListView 
+channelListHeaderViewModel.bindView(chanelHeaderView, viewLifecycleOwner)
+// Optional: setup listeners for user avatar and action button 
+chanelHeaderView.setOnActionButtonClickListener {
+    navigateSafely(R.id.action_homeFragment_to_addChannelFragment)
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:33]: Hm, `navigateSafely` is something only relevant to our sample app. IMO, better just to leave a comment that you can navigate to create channel screen here.

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md [hunk 10]
```diff
@@ -1,17 +1,51 @@
...
+}
+chanelHeaderView.setOnUserAvatarClickListener {
+    binding.drawerLayout.openDrawer(GravityCompat.START)
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md:36]: Same, seems that nav drawer is something that is only relevant to our sample app

**Review by bychkovdmitry — APPROVED** [review]

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md (hunk 9 above)
Comment (samiuelson) [reply docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md]: Fixed

File: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md (hunk 10 above)
Comment (samiuelson) [reply docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md]: Fixed

**Pull request merged**